src/solvers/imex_ssprk.jl

```diff
@@ -60,7 +61,7 @@ function step_u!(integrator, cache::IMEXSSPRKCache)
     (; U, U_lim, U_exp, T_lim, T_exp, T_imp, temp, β, γ, newtons_method_cache) = cache
     s = length(b_imp)
 
-    if !isnothing(T_imp!)
+    if !isnothing(T_imp!) && !isnothing(newtons_method)
```

Comment:
Are there cases when `!isnothing(T_imp!)` and `isnothing(newtons_method)`? Or is this just an extra safety guard?

Reply:
The ported explicit algorithms run using code, but without a Newton's method. This check is needed to avoid allocating a Newton's method for an explicit algorithm.